### PR TITLE
Add BUILD_STATIC option to create static libraries

### PR DESCRIPTION
**Pull request by drdanz**

### CODE REVIEW

**Review by iamsergio — APPROVED** [review]

**Comment by dantti** [comment]
This should use option BUILD_SHARED_LIBS instead.

File: CMakeLists.txt [hunk 1]
```diff
@@ -76,6 +79,13 @@ else()
     option(BUILD_TESTS "Build the test harness" ON)
 endif()
 option(BUILD_QT6 "Build against Qt 6" OFF)
+option(BUILD_STATIC "Build statically" OFF)
```
Comment (dantti) [inline CMakeLists.txt:82]: option(BUILD_SHARED_LIBS "Build in shared lib mode" ON)

File: CMakeLists.txt [hunk 2]
```diff
@@ -76,6 +79,13 @@ else()
...
+
+if(BUILD_STATIC)
```
Comment (dantti) [inline CMakeLists.txt:84]: then get rid of

File: src/core/CMakeLists.txt [hunk 3]
```diff
@@ -77,7 +77,7 @@ if(GRAPHVIZ_FOUND)
     endif()
 endif()
 
-add_library(kdstatemachineeditor_core SHARED ${LIB_SRCS})
+add_library(kdstatemachineeditor_core ${BUILD_LIBRARY_MODE} ${LIB_SRCS})
```
Comment (dantti) [inline src/core/CMakeLists.txt:80]: also not needed here just add_library(kdstatemachineeditor_core ${LIB_SRCS})

**Review by dantti — COMMENTED** [review]
After those changes I recommend using generate_export_header otherwise having the exports right for Windows requires lots of manual updates to ifdefs

**Pull request merged**